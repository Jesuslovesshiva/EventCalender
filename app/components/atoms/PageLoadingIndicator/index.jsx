import React from 'react'

import Spinner from '@/components/atoms/Spinner'

const PageLoadingIndicator = React.forwardRef((props, ref) => (
  <div ref={ref} className="w-full h-full flex justify-center items-center">
    <Spinner className="p-4 m-4" />
  </div>
))

PageLoadingIndicator.displayName = 'PageLoadingIndicator'

export default React.memo(PageLoadingIndicator)
