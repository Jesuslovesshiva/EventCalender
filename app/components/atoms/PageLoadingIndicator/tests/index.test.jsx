import render from '@/utils/test-utils/custom-render'

import PageLoadingIndicator from '../index'

const renderComponent = () => render(<PageLoadingIndicator />)

describe('PageLoadingIndicator', () => {
  it('should match the snapshot', () => {
    const {container} = renderComponent()
    expect(container.firstChild).toMatchSnapshot()
  })
})
