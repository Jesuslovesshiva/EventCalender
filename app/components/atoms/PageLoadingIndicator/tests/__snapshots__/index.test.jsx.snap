// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLoadingIndicator should match the snapshot 1`] = `
<div
  class="w-full h-full flex justify-center items-center"
>
  <div
    class="animate-spin rounded-full w-8 h-8 border-4 border-primary/20 border-l-primary p-4 m-4"
    role="status"
  >
    <span
      class="sr-only"
    >
      Wird geladen...
    </span>
  </div>
</div>
`;
